src/navigation.ts

```diff
@@ -537,14 +537,34 @@ export class Navigation {
     this.setCursorOnWorkspaceFocus(workspace, keepCursorPosition);
   }
 
+  /**
+   * Blurs (de-focuses) the workspace's toolbox or flyout, and hides the flyout
+   * if it's currently visible.
+   *
+   * @param workspace The workspace containing the toolbox or flyout.
+   */
+  blurToolboxAndFlyout(workspace: Blockly.WorkspaceSvg) {
+    workspace.hideChaff();
+    const reset = !!workspace.getToolbox();
+
+    this.resetFlyout(workspace, reset);
+    switch (this.getState(workspace)) {
+      case Constants.STATE.FLYOUT:
+      case Constants.STATE.TOOLBOX:
+        // Clear state since neither the flyout nor toolbox are focused anymore.
+        this.setState(workspace, Constants.STATE.NOWHERE);
```

Comment:
Will this ever be called when state needs to move directly to the workspace, instead of to `NOWHERE`?

Reply:
Not directly from what I've observed. The logic here is largely relying on the focus/blur events to be handled in the correct order at the DOM level. When tab navigating from a focused toolbox to the workspace this method will be called to reset state to nowhere, then the workspace will receive focus and this state will immediately be updated to be workspace.

We needed an intermediary state because the assumption that workspace is always 'next' is also false (when shift tabbing from the toolbox).

I'm open to suggestions on alternate designs or ways of making this flow a bit clearer since I recognize that it is a bit obscured currently.

Reply:
> The logic here is largely relying on the focus/blur events to be handled in the correct order at the DOM level. When tab navigating from a focused toolbox to the workspace this method will be called to reset state to nowhere, then the workspace will receive focus and this state will immediately be updated to be workspace.

Got it. Good to keep in mind as a constraint, in case we later find that the events are coming in in a different order in some configurations, but this explanation makes sense to me.